source/usage/other/craftscripts.rst

```diff
@@ -47,54 +45,128 @@ Scripts have the following three variables in their global namespace:
+
+.. topic:: Example: Importing packages and classes
+
+    ::
+
+        // Equivalent to `import com.sk89q.worldedit.world.*`
+        importPackage(Packages.com.sk89q.worldedit.world);
+        // Equivalent to `import com.sk89q.worldedit.world.block.*`
+        importPackage(Packages.com.sk89q.worldedit.world.block);
+        // Equivalent to `import java.utils.*`
+        importPackage(java.util)
+
+        // Equivalent to`import com.sk89q.worldedit.math.BlockVector3`
+        importClass(Packages.com.sk89q.worldedit.math.BlockVector3);
+        // Equivalent to `import java.utils.ArrayList`
+        importClass(java.util.ArrayList)
+
+Warning: unlike Java, Rhino JavaScript does not implicitly import ``java.lang.*``, and doing so explicitly is not permitted.
+
+Working with Blocks
 -------------------
 
-All block editing in WorldEdit is done through an EditSession. This object handles history and block placement order all automatically. To get an edit session for your own script, use:
+All block editing in WorldEdit is done through an EditSession. This object handles history and block placement order automatically. To get an edit session for your own script, use:
 
 ::
 
-    var sess = context.remember();
+    var editSession = context.remember();
 
-Every time you call that method, you will get a new ``EditSession``, so be sure to keep one around. To set blocks, you will either need to provide a ``BlockState`` which is a combination of a block type and one or more states, or a ``BaseBlock``, which is a ``BlockState`` that may additionally have NBT data.
+This returns a new instance of `EditSession <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java>`_. To set blocks, you will need to provide a `BlockVector3 <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/math/BlockVector3.java>`_ and then either a `BlockState <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockState.java>`_ or a `BaseBlock <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BaseBlock.java>`_ to the ``EditSession``.
 
-.. topic:: Example: Setting a block to white wool
+
+.. topic:: Example: Setting white wool at (431, 63, -41)
 
     ::
 
-        importPackage(Packages.com.sk89q.worldedit.world.block);
+        importClass(Packages.com.sk89q.worldedit.world.block.BlockTypes);
+        importClass(Packages.com.sk89q.worldedit.math.BlockVector3);
+
+        var editSession = context.remember();
+        editSession.setBlock(BlockVector3.at(431, 63, -41), BlockTypes.WHITE_WOOL.getDefaultState());
+
+
+
+To get blocks, use the function ``EditSession#getBlock``, which accepts a ``BlockVector3`` and returns an instance of ``BaseBlock``.
```

Comment:
This is incorrect, `getBlock` only returns `BlockState`, you need `getFullBlock` to retrieve a `BaseBlock`.

Reply:
Fixed in https://github.com/EngineHub/WorldEditDocs/pull/13/commits/f4440bf5baec9d16939e8b9c508a5fc9d841fc7b